pkg/provider/configuration.go

```diff
@@ -353,7 +353,7 @@ func BuildTCPRouterConfiguration(ctx context.Context, configuration *dynamic.TCP
 			if len(configuration.Services) > 1 {
 				delete(configuration.Routers, routerName)
 				loggerRouter.
-					Error("Could not define the service name for the router: too many services")
+					Errorf("No service name for the router: %s", routerName)
```

Comment:
I understand it's not common practice, but a direct link to the docs would be helpful here. When you're changing a message that is hit regularly, losing the google-juice makes it difficult to move forward from the error. In order of priority for handling an error I would choose:
1. Don't error, just fix it (this was the approach in the other PR)
2. Give the exact info that is necessary to fix the problem
3. Link to the exact info that is necessary
4. Describe the error (we're here)

Reply:
With ceed9c579, I added a link in the doc to the error message.

Reply:
BTW, I agree with you on 1. :)